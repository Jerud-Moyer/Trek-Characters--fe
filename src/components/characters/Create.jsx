import React from 'react';
import PropTypes from 'prop-types';
import styles from './Create.css';

const Create = ({
  name,
  affiliation,
  origin,
  race,
  imageUrl,
  onChange,
  onSubmit
}) => (
  <div className={styles.formDiv}>
    <form onSubmit={onSubmit} className={styles.form}>
      <label htmlFor="name">Name</label>
      <input
        id="name"
        type="text"
        name="name"
        value={name}
        onChange={onChange}
      />
      <label htmlFor="affiliation">Affiliation</label>
      <input
        id="affiliation"
        type="text"
        name="affiliation"
        value={affiliation}
        onChange={onChange}
      />
      <label htmlFor="origin">Origin</label>
      <input
        id="origin"
        type="text"
        name="origin"
        value={origin}
        onChange={onChange}
      />
      <label htmlFor="race">Race</label>
      <input
        id="race"
        type="text"
        name="race"
        value={race}
        onChange={onChange}
      />
      <label htmlFor="imageUrl">Image url</label>
      <input
        id="imageUrl"
        type="url"
        name="imageUrl"
        value={imageUrl}
        onChange={onChange}
      />
      <button>Submit</button>
    </form>

  </div>
);

Create.propTypes = {
  name: PropTypes.string.isRequired,
  affiliation: PropTypes.string.isRequired,
  origin: PropTypes.string.isRequired,
  race: PropTypes.string.isRequired,
  imageUrl: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Create;
