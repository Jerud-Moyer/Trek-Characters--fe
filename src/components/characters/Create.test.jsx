import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Create from './Create';

describe('Create component', () => {
  afterEach(() => cleanup());

  it('renders Character', () => {
    const { asFragment } = render(<Create
      name="test Name"
      affiliation="Saturn"
      origin="pluto"
      race="Uranusonian"
      imageUrl="Test Image"
      onChange={() => {}}
      onSubmit={() => {}}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
