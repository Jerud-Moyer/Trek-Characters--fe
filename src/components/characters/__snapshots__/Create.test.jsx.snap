// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create component renders Character 1`] = `
<DocumentFragment>
  <div
    class="formDiv"
  >
    <form
      class="form"
    >
      <label
        for="name"
      >
        Name
      </label>
      <input
        id="name"
        name="name"
        type="text"
        value="test Name"
      />
      <label
        for="affiliation"
      >
        Affiliation
      </label>
      <input
        id="affiliation"
        name="affiliation"
        type="text"
        value="Saturn"
      />
      <label
        for="origin"
      >
        Origin
      </label>
      <input
        id="origin"
        name="origin"
        type="text"
        value="pluto"
      />
      <label
        for="race"
      >
        Race
      </label>
      <input
        id="race"
        name="race"
        type="text"
        value="Uranusonian"
      />
      <label
        for="imageUrl"
      >
        Image url
      </label>
      <input
        id="imageUrl"
        name="imageUrl"
        type="url"
        value="Test Image"
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</DocumentFragment>
`;
